substrate/client/state-db/src/noncanonical.rs

```diff
@@ -189,8 +201,15 @@ impl<BlockHash: Hash, Key: Hash> NonCanonicalOverlay<BlockHash, Key> {
 			let mut total: u64 = 0;
 			block += 1;
 			loop {
-				let mut level = OverlayLevel::new();
-				for index in 0..MAX_BLOCKS_PER_LEVEL {
+				let level_span = db
+					.get_meta(&to_meta_key(OVERLAY_LEVEL_SPAN, &block))
+					.map_err(Error::Db)?
```

Comment:
Wouldn't that break compatibility with existing databases taht don't have the `OVERLAY_LEVEL_SPAN` key?

Reply:
I think `MetaDb::get_meta` in case of a missing key returns `None`, right? If so, this will default to iterating through indices `0..OVERLAY_LEVEL_STORE_SPANS_LONGER_THAN`, which is the same as the old behavior.